contents/docs/session-replay/_snippets/ios-logging.mdx

```diff
@@ -0,0 +1,96 @@
+
+```ios_swift
+let config = PostHogConfig(apiKey: <ph_project_api_key>, host: <ph_client_api_host>)
+config.sessionReplay = true
+config.sessionReplayConfig.captureLogs = true // TIP: Needs to be configured in code. Project settings won't have an effect.
+
+PostHogSDK.shared.setup(config)
+```
+
+> **Important:** Individual console logs are truncated at 2000 characters. The rest of the log is not sent to PostHog. When truncating a log message we add ...[Truncated] to the message.
+
+### Custom log processing and sanitization
+
+PostHog iOS SDK allows you to customize how log messages are processed and categorized by providing a custom `processConsoleOutput` handler. 
+This handler receives raw console output and returns an array of processed log entries with their respective severity levels:
+
+```ios_swift
+let config = PostHogConfig(apiKey: <ph_project_api_key>, host: <ph_client_api_host>)
+
+// Custom log processing and sanitization
+config.sessionReplayConfig.captureLogsConfig.processConsoleOutput = { output in
+
+    // Skip trace-level logs
+    guard !output.contains("TRACE") else {
+        return []
+    }
+
+    // Determine log level based on message content
+    let level: PostHogConsoleLogLevel = {
+        if output.contains("CRITICAL") { return .error }
+        if output.contains("WARN") { return .warn }
+        return .info
+    }()
+    
+    // Remove sensitive information (e.g., auth tokens)
+    let sanitizedMessage = output.replacingOccurrences(
+        of: #"Bearer\s+[A-Za-z0-9-._~+/]+=*"#,
+        with: "Bearer [TOKEN REDACTED]",
+        options: .regularExpression
+    )
+    
+    return [PostHogConsoleLogResult(level: level, message: sanitizedMessage)]
+}
+
+PostHogSDK.shared.setup(config)
+```
+
+### Configuring minimum log level
+
+By default, the SDK only captures error-level logs to minimize noise in session replays.
+ You can adjust this using the `minimumLogLevel` setting:
+
+```ios_swift
+// Capture errors only (default)
+config.sessionReplayConfig.captureLogsConfig.minimumLogLevel = .error
+
+// Capture warnings and errors
+config.sessionReplayConfig.captureLogsConfig.minimumLogLevel = .warn
+
+// Capture all logs (most verbose)
+config.sessionReplayConfig.captureLogsConfig.minimumLogLevel = .info
+```
+
+Only logs with a level greater than or equal to the `minimumLogLevel` will be captured. For example, if set to `.warn`, both **warning** and **error** logs will be captured, but **info** logs will be filtered out.
+
+
+### Default behavior
+
+#### Log level
+The SDK captures raw console output directly, independent of any specific logging framework or implementation. 
+Since the raw output doesn't inherently carry log level information, the SDK deduces severity levels by analyzing the log content.
+By default, the SDK automatically classifies log messages into three severity levels:
```

Comment:
does this match the filters we support at posthog UI?

Reply:
yep we defuckulate various (browser) levels down into those three

https://github.com/PostHog/posthog/blob/16711c3e294fbf23ba7ae048db7202d50b34d049/plugin-server/src/main/ingestion-queues/session-recording/process-event.ts#L81-L105